docs/src/main/tut/TypedDataset.md

```diff
@@ -1 +1,158 @@
+import org.apache.spark.sql.SparkSession
+
+val conf = new SparkConf().setMaster("local[*]").setAppName("test").set("spark.ui.enabled", "false").set("spark.app.id", "tut-dataset")
+val spark = SparkSession.builder().config(conf).getOrCreate() 
+
+System.clearProperty("spark.master.port")
+System.clearProperty("spark.driver.port")
+System.clearProperty("spark.hostPort")
+System.setProperty("spark.cleaner.ttl", "300")
+
+// We are using this directory so let's make sure it is clean first 
+org.apache.commons.io.FileUtils.deleteDirectory(new java.io.File("/tmp/foo/"))
+```
+
+## Comparing Frameless TypedDatasets with standard Spark Datasets
+
+
+**Goal:** 
+  This tutorial compares the standard Spark Datasets api with the one provided by
+  frameless' TypedDataset. It shows how TypedDatsets allows for an expressive and 
+  typesafe api with no compromises on performance. 
+ 
+ 
+For this tutorial we first create a simple dataset and save it on disk as a parquet file. 
+[Parquet](https://parquet.apache.org/) is a popular columnar format and well supported by Spark. 
+It's important to note that when operating on parquet datasets, Spark knows that each column is stored 
+separately, so if we only need a subset of the columns Spark will optimize for this and avoid reading 
+the entire dataset. This is a rather simplistic view of how Spark and parquet work together but it
+will serve us well for the context of this discussion.
+
+```tut:book
+import spark.implicits._
+
+// Our example case class Foo acting here as a schema
+case class Foo(i: Long, j: String)
+
+// Assuming spark is loaded and SparkSession is bind to spark
+val initialDs = spark.createDataset( Foo(1, "Q") :: Foo(10, "W") :: Foo(100, "E") :: Nil )
+
+// Assuming you are on Linux or Mac OS
+initialDs.write.parquet("/tmp/foo")
+
+val ds = spark.read.parquet("/tmp/foo").as[Foo]
+
+ds.show()
+```
+
+The value `ds` holds the content of the `initialDs` read from a parquet file. 
+Let's try to only use field `i` from Foo and see how Spark's Catalyst (the query optimizer) 
+optimizes this. 
+
+```tut:book
+// Using a standard Spark TypedColumn in select()
+val filteredDs = ds.filter($"i" === 10).select($"i".as[Long])
+
+filteredDs.show()
+```
+
+The `filteredDs` is of type `Dataset[Long]`. Since we only access field `i` from `Foo` the type is correct.
+Unfortunately, this syntax requires handholding by explicitly setting the `TypedColumn` in the `select` statement
+to return type `Long` (look at the `as[Long]` statement). We will discuss this limitation next in more detail. 
+Now, let's take a quick look at the optimized Physical Plan that Spark's Catalyst generated.
+
+```tut:book
+
+filteredDs.explain()
+
+```
+
+The last line is very important (see `ReadSchema`). The schema read 
+from the parquet file only required reading column `i` without needing to access column `j`.
+This is great! We have both an optimized query plan and typesafety! 
+
+Unfortunately, this syntax is not bulletproof. 
```

Comment:
I would simplify this paragraph to the following:

Unfortunately, this syntax is not bulletproof: it fails at run-time if we try to access a non existing column `x`: